src/main/java/org/opentripplanner/routing/edgetype/SimpleTransfer.java

```diff
@@ -39,8 +39,11 @@ public SimpleTransfer(TransitStop from, TransitStop to, double distance, LineStr
     @Override
     public State traverse(State s0) {
         // only allow to use TransferEdges when already on a transit vehicle and not on a street level
-        if (!(s0.backEdge instanceof PreAlightEdge)) {
-            return null;
+        if (s0.getOptions().arriveBy) {
+            if (s0.getBackEdge() instanceof TransferEdge || s0.getBackEdge() instanceof SimpleTransfer) return null;
```

Comment:
Is an equivalent for the front/forward edge that we can check?

Reply:
Yep, added this. Did some testing and results look good.

E.g. for the [newly reported issue](https://mbtace.slack.com/archives/C911LQH2N/p1559752666005000): 

**Before**:
![image](https://user-images.githubusercontent.com/45011335/58982880-7e4e7200-87a3-11e9-9a4c-9665e3537f1b.png)

**After**:
![image](https://user-images.githubusercontent.com/45011335/58982904-89090700-87a3-11e9-8775-9b83e1907e4d.png)